posthog/temporal/batch_exports/snowflake_batch_export.py

```diff
@@ -426,6 +456,13 @@ async def amerge_person_tables(
         person_distinct_id_version_key: str = "person_distinct_id_version",
     ):
         """Merge two identical person model tables in Snowflake."""
+
+        # handle the case where the final table doesn't contain all the fields present in the stage table
+        # (for example, if we've added new fields to the person model)
+        final_table_columns = await self.aget_table_columns(final_table)
+        final_table_column_names = [field[0] for field in final_table_columns]
+        update_when_matched = [field for field in update_when_matched if field[0] in final_table_column_names]
```

Comment:
Is there a chance we do not have permissions to `SELECT` from the final table but we do have permissions to `INSERT` (or `MERGE`) into it?

Reply:
Especially considering we are not `SELECT`-ing from final table but rather `information_schema.columns`. I think we may need to be defensive about permissions here.

Reply:
Good point. Maybe there's a way to get the column information in a similar way to Postgres & Redshift. Will check